### PR TITLE
honor VSO_DEDUP_REDIRECT_TIMEOUT_IN_SEC

### PR DESCRIPTION
**Pull request by johnterickson**
Should help mitigate https://github.com/microsoft/azure-pipelines-tasks/issues/11864 when there is TCP reliability issues.
Example output:
```
Resolved to: tools-1
##[debug]Processed: ##vso[task.settaskvariable variable=RESTORE_STEP_RAN;issecret=False;]true
##[debug]Processed: ##vso[task.settaskvariable variable=RESTORE_STEP_RESOLVED_FINGERPRINT;issecret=False;]tools-1
Set VSO_DEDUP_REDIRECT_TIMEOUT_IN_SEC to `5`.
Overriding default max parallelism with 256
Max dedup parallelism: 256
...
Information, Downloaded 35.0 MB out of 2,649.5 MB (1%).
Information, [https://zrtvsblobprodeus2154.blob.core.windows.net/dbb7ce0259fa64471f91ca8f195e3bc60c/5043FF8213DD0B7EE17C06DF799E18061B8EA1399DA09D9BDD5B9A92D2E4AB8901?sv=2019-07-07&sr=b&spr=https&se=2022-12-03T19:18:48Z&sp=r&rscl=x-e2eid-8d6a7ea4-1f1244ef-8f73ee5f-f65a9fb5-session-371df360-90564abf-95c1419b-2382ca37] Try 1/5, Task was requested to be canceled.
Verbose, [https://zrtvsblobprodeus2154.blob.core.windows.net/REDACTED] Try 1/5, retryable exception caught. Retrying in 00:00:01. Microsoft.VisualStudio.Services.Content.Common.AsyncHttpRetryHelper+RetryableException: Attempt timed out after 00:00:05.
 ---> System.OperationCanceledException: The operation was canceled.
   at System.Threading.CancellationToken.ThrowOperationCanceledException()
   at System.Threading.CancellationToken.ThrowIfCancellationRequested()
   at Microsoft.VisualStudio.Services.Content.Common.AsyncHttpRetryHelper`1.InvokeAsync(CancellationToken cancellationToken)
   at Microsoft.VisualStudio.Services.BlobStore.WebApi.RequestHandler.<>c__DisplayClass18_0.<<HandleRedirectAsync>b__0>d.MoveNext()
   --- End of inner exception stack trace ---
   at Microsoft.VisualStudio.Services.BlobStore.WebApi.RequestHandler.<>c__DisplayClass18_0.<<HandleRedirectAsync>b__0>d.MoveNext()
--- End of stack trace from previous location where exception was thrown ---
   at Microsoft.VisualStudio.Services.Content.Common.AsyncHttpRetryHelper`1.InvokeAsync(CancellationToken cancellationToken)
Verbose, Details:
No LastRequestResponse on exception RetryableException: Attempt timed out after 00:00:05.

```

### CODE REVIEW

File: src/Microsoft.VisualStudio.Services.Agent/Blob/DedupManifestArtifactClientFactory.cs [hunk 1]
```diff
@@ -188,6 +190,26 @@ public int GetDedupStoreClientMaxParallelism(AgentTaskPluginExecutionContext con
             return parallelism;
         }
 
+        private static readonly string[] EnvironmentVariables = new [] { "VSO_DEDUP_REDIRECT_TIMEOUT_IN_SEC" };
```
Comment (JohnSchmeichel) [inline src/Microsoft.VisualStudio.Services.Agent/Blob/DedupManifestArtifactClientFactory.cs:193]: did we move away from the VSO_IM_ON_A_BOAT naming?

**Review by JohnSchmeichel — APPROVED** [review]

**Review by carl-tanner — COMMENTED** [review]
Looks good.

**Review by carl-tanner — APPROVED** [review]

**Comment by carl-tanner** [comment]
.azp run

**Comment by johnterickson** [comment]
Some test runs:
image

**Comment by johnterickson** [comment]
image

**Review by KonstantinTyukalov — APPROVED** [review]

**Review by LiliaSabitova — APPROVED** [review]

**Comment by shekibobo** [comment]
When could users expect to start seeing improvements in the cache task?

**Comment by pweiss-mri** [comment]
Any news on the release?  It's been in limbo for a long time

**Comment by johnterickson** [comment]
Release is rolled out!

**Comment by blyscuit** [comment]
Where do we put the env `VSO_DEDUP_REDIRECT_TIMEOUT_IN_SEC`
Is it with cache task?
```
    - task: Cache@2
      inputs:
        key: 'pods | "$(Agent.OS)" | Podfile.lock'
        path: 'Pods'
        cacheHitVar: 'PODS_CACHE_RESTORED'
      env:
        VSO_DEDUP_REDIRECT_TIMEOUT_IN_SEC: 5
      displayName: Cache Pods
```

**Comment by shekibobo** [comment]
> You will need to set this to opt-in to the change. If you are on a weaker network connection you may want to increase the timeout.
> 
> ```
> env:
>   VSO_DEDUP_REDIRECT_TIMEOUT_IN_SEC: 5
> ```
> 
> The agent had rolled back because of a regression, but it looks like it is rolling forward again. It's usually rolled out over the course of ~days. It will be `214.1`.

When you say `2.214.1`, do you mean a new version of the cache task? I'm still seeing version `2.198.0` of the cache task on my azure hosted mac agents, and my caches are still timing out a lot even with the environment variable set. Is there some way I can ensure the latest build image?

**Comment by adrian-skybaker** [comment]
>You will need to set this to opt-in to the change. If you are on a weaker network connection you may want to increase the timeout.

Do both of these statements apply to the MS hosted agents?

**Comment by pweiss-mri** [comment]
@adrian-skybaker
> Do both of these statements apply to the MS hosted agents?

Yes these apply to MS hosted agents.  I'm using MS hosted agents and saw massive performance improvements by setting a 5 second timeout for just the `Cache@2` tasks

**Comment by mxa0079** [comment]
Is this fix agnostic to the version of the cache task?

**Pull request merged**